crates/mun_codegen/src/code_gen/linker.rs

```diff
@@ -17,137 +13,101 @@ pub fn create_with_target(target: &spec::Target) -> Box<dyn Linker> {
 pub trait Linker {
     fn add_object(&mut self, path: &Path);
     fn build_shared_object(&mut self, path: &Path);
-    fn finalize(&mut self) -> process::Command;
+    fn finalize(&mut self) -> Result<(), String>;
 }
 
 struct LdLinker {
-    cmd: process::Command,
-}
-
-struct Ld64Linker {
-    cmd: process::Command,
-}
-
-struct MsvcLinker {
-    cmd: process::Command,
+    args: Vec<String>,
 }
 
 impl LdLinker {
     fn new(_target: &spec::Target) -> Self {
-        let mut cmd = process::Command::new(LLD_PATH.as_os_str());
-        cmd.arg("-flavor");
-        cmd.arg("ld");
-
-        LdLinker { cmd }
-    }
-}
-
-impl Ld64Linker {
-    fn new(target: &spec::Target) -> Self {
-        let mut cmd = process::Command::new(LLD_PATH.as_os_str());
-        cmd.arg("-flavor");
-        cmd.arg("ld64");
-
-        cmd.arg("-arch");
-        cmd.arg(&target.arch);
-
-        Ld64Linker { cmd }
-    }
-}
-
-lazy_static::lazy_static! {
-    static ref LLD_PATH: PathBuf = {
-        let path = std::env::current_exe()
-            .unwrap()
-            .parent()
-            .unwrap()
-            .to_path_buf();
-
-        let mut binary_dir:&Path = &path;
-        let mut binary_path;
-        loop {
-            binary_path = binary_dir.to_path_buf();
-            binary_path.push(format!("lld{}", std::env::consts::EXE_SUFFIX));
-            if binary_path.exists() {
-                break;
-            }
-            binary_dir = binary_dir.parent().expect("could not find lld");
+        LdLinker {
+            args: Vec::default(),
         }
-        binary_path
-    };
-}
-
-impl MsvcLinker {
-    fn new(_target: &spec::Target) -> Self {
-        let mut cmd = process::Command::new(LLD_PATH.as_os_str());
-        cmd.arg("-flavor");
-        cmd.arg("link");
-
-        MsvcLinker { cmd }
     }
 }
 
 impl Linker for LdLinker {
     fn add_object(&mut self, path: &Path) {
-        self.cmd.arg(path);
+        self.args.push(path.to_string_lossy().to_string());
```

Comment:
`to_string_lossy()` can silently introduce errors. We could do `to_str().ok_or(...)`